File: backend/voting-app/src/main/java/com/tu/votingapp/entities/RegionEntity.java
package com.tu.votingapp.entities;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@EqualsAndHashCode(callSuper=false)
@AllArgsConstructor
@NoArgsConstructor
public class RegionEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @NonNull
    @Column(name = "region_name", nullable = false)
    private String name;

    @Column(name = "population")
    private Integer population;

    public RegionEntity(Long id) {
        this.id = id;
    }
}

File: frontend/android/app/src/main/java/com/example/android/entity/RegionEntity.kt
package com.example.android.entity

import com.example.android.db.orm.Column
import com.example.android.db.orm.Table

@Table(name = "Region")
data class RegionEntity(
    @Column(name = "id", primaryKey = true, autoIncrement = true)
    val id: Long? = null,

    @Column(name = "name")
    val name: String,

    @Column(name = "population")
    val population: Int
)